LiteCore/Support/LogDecoder.cc

```diff
@@ -218,8 +217,8 @@ namespace litecore {
                         case 'd':
                         case 'i':
                             {
-                                bool    negative = _in.get() > 0;
-                                int64_t param    = readUVarInt();
+                                bool negative = _in.get() > 0;
+                                auto param    = static_cast<int64_t>(readUVarInt());
```

Comment:
Are these in need of narrow_cast?

Reply:
Ya, narrow_cast will do runtime check. Don't know what static_cast will do to convert unsigned to signed. How will it be different from plain assignment?

Reply:
Plain assignment (implicit cast) is platform-dependent. I can change it to narrow_cast if we think that is more suitable.

Reply:
My understanding is implicit cast and static_cast do the same: keep all the bits unchanged. All platforms use 2's complement to represent negative.

Reply:
From LLVM's documentation:
> The C/C++ standard does not mandate two’s complement for signed integers, and so the compiler is free to define what the semantics are for converting an unsigned integer to signed integer. Clang’s implementation uses the two’s complement format.

Reply:
And yeah, implicit cast and static_cast do the same thing. Just implicit cast means it's not obvious that you intended to cast between signed/unsigned and therefore you aren't necessarily aware of the dangers. Whereas static_cast makes it explicit that you understand you are converting between signed/unsigned and accept the dangers.

Reply:
Basically static cast only does some trivial check that you are assigning to compatible types.  For example you cannot static cast from int to string.  It's marginally better than implicit cast (which in C++ is called reinterpret_cast) which literally just takes the size of the underlying struct or pointer and reads it as another.

narrow_cast in release mode is just static_cast, but in debug mode it checks that the result is not out of bounds.